Source/NonlinearSolvers/PicardSolver.H

```diff
@@ -0,0 +1,214 @@
+    {
+        a_rtol = m_rtol;
+        a_atol = m_atol;
+        a_maxits = m_maxits;
+    }
+
+    void PrintParams () const override
+    {
+        amrex::Print() << "Picard max iterations:      " << m_maxits << std::endl;
+        amrex::Print() << "Picard relative tolerance:  " << m_rtol << std::endl;
+        amrex::Print() << "Picard absolute tolerance:  " << m_atol << std::endl;
+        amrex::Print() << "Picard require convergence: " << (m_require_convergence?"true":"false") << std::endl;
+    }
+
+    void Verbose ( bool  a_verbose ) override { m_verbose = a_verbose; }
+
+protected:
+
+    bool m_is_defined = false;
+    bool m_verbose = true;
+    bool m_require_convergence = true;
+
+    mutable Vec m_Usave, m_R;
+    Ops* m_ops = nullptr;
+
+/**
+ * \brief relative tolerance for the Picard nonlinear solver
+ */
+    amrex::Real m_rtol = 1.0e-6;
+
+/**
+ * \brief absolute tolerance for the Picard nonlinear solver
+ */
+    amrex::Real m_atol = 0.;
+
+/**
+ * \brief maximum iterations for the Picard nonlinear solver
+ */
+    int m_maxits = 100;
+
+private:
+
+    void ParseParameters( );
+
+};
+
+template <class Vec, class Ops>
+void PicardSolver<Vec,Ops>::Define ( const Vec&  a_U,
+                                     Ops*        a_ops )
+{
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+        !m_is_defined,
+        "Picard nonlinear solver object is already defined!");
+
+    ParseParameters();
+
+    m_Usave.Define(a_U);
+    m_R.Define(a_U);
+
+    m_ops = a_ops;
+
+    m_is_defined = true;
+
+}
+
+template <class Vec, class Ops>
+void PicardSolver<Vec,Ops>::ParseParameters ()
+{
+    const amrex::ParmParse pp_picard("picard");
+    pp_picard.query("verbose",             m_verbose);
+    pp_picard.query("absolute_tolerance",  m_atol);
+    pp_picard.query("relative_tolerance",  m_rtol);
+    pp_picard.query("max_iterations",      m_maxits);
+    pp_picard.query("require_convergence", m_require_convergence);
+
+}
+
+template <class Vec, class Ops>
+void PicardSolver<Vec,Ops>::Solve ( Vec&         a_U,
+                              const Vec&         a_b,
+                                    amrex::Real  a_time,
+                                    amrex::Real  a_dt ) const
+{
+    BL_PROFILE("PicardSolver::Solve()");
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+        m_is_defined,
+        "PicardSolver::Solve() called on undefined object");
+    using namespace amrex::literals;
+
+    //
+    // Picard fixed-point iteration method to solve nonlinear
+    // equation of form: U = b + R(U)
+    //
+
+    amrex::Real norm_abs = 0.;
+    amrex::Real norm0 = 1._rt;
+    amrex::Real norm_rel = 0.;
+
+    int iter;
+    for (iter = 0; iter < m_maxits;) {
+
+        // Save previous state for norm calculation
+        m_Usave = a_U;
+
+        // Update the solver state (a_U = a_b + m_R)
+        m_ops->PreRHSOp( a_U, a_time, a_dt, iter, false );
+        m_ops->ComputeRHS( m_R, a_U, a_time, a_dt );
+        a_U = a_b;
+        a_U += m_R;
```

Comment:
The cost of extra multiplication involving a scalar is usually minimal. But this is not a LinComb (which involves three MultiFabs). This can be done with `MultiFab::Add`.